### DISCUSSION TITLE
Seq does not handover all incoming events to app

### DISCUSSION THREAD

**Discussion by apatenge-bhs** [datalust/seq-tickets#2307] [Help] [answered]
I have an application which is feeding in around 150000 events in 5 minutes to Seq. These events appear in seq as they should do. 

But not all events are handed over to my own developed Seq App. I added a simple counter there and it seems 1/4 of the events are not handed over there. So it seems Seq is not calling the "public async Task OnAsync(Event<LogEvent> evtLogEvent)" method.

What can be the issue here?

**Comment by nblumhardt** [comment]
Hi @apatenge, thanks for your message.

The most likely cause will be your app not processing the events fast enough; Seq has an internal buffer per app of 100000 events, and if the app falls behind, events will be dropped.

An event is normally written to Seq's internal diagnostic log when this happens - check out Settings > Diagnostics > Download a full diagnostic report, and search for `Capacity of {Capacity} events exceeded by channel`. The message is only written once per app per day, so if the log contained in the report doesn't cover a full day, you might miss it.

You can work around this by speeding up the app, but if that isn't possible, you can also modify the implementation of the app to buffer the events internally using some kind of queue, and process them on a second/background thread. Memory use will need to be carefully monitored if you take this path, however.

Let me know if this helps,
Nick

**Reply by apatenge-bhs** [reply]
Hello Nick,

many thanks for that fast answer. That explains exactly what happens. We have in reality 150000 events per hour. That are 24 ms per event. My app takes ca. 7 ms per event to process these. This is working as expected. To save time we want to simulate the events. Therefore we are loading these in ca. 5 minutes into Seq. There we have just 2ms per event to process. So the app falls behind and events are dropped.

The question is, can we increase the buffer from 100000 to maybe 250000 events? That would solve the problem most probably quickly.

I will also check the option with the own buffer in my app. But to buffer the event in 2ms will be also short. 

Many thanks,
André

**Reply by nblumhardt** [reply, marked as answer]
Hi André, thanks for your reply. The buffer size isn't configurable - waiting a few ms between events will be the way to go, currently.

Best regards,
Nick
